kubernetes/resource_kubernetes_env.go

```diff
@@ -60,6 +60,12 @@ func resourceKubernetesEnv() *schema.Resource {
 				Description: "Name of the container for which we are updating the environment variables.",
 				Required:    true,
 			},
+			"init_container": {
```

Comment:
The options we discussed on our call today:

1. Don't add a new attribute, and just use `container` and find the container with that name in initContainers. Container names have to be unique across initContainers and containers so this should work.

2. To be explicit, add a `init_container` attribute which is a string and functions the same as `container` but only applies to the initContainers list. We can use `ExactlyOneOf` to enforce that one of `container` or `init_container` must be set. This would make the configuration a bit more explicit in it's intentions.